baremaps-openapi/src/main/java/com/baremaps/openapi/services/CollectionsService.java

```diff
@@ -16,33 +16,182 @@
+
+  static ObjectMapper mapper = new ObjectMapper();
+
+  @Inject
+  public CollectionsService(Jdbi jdbi) {
+    this.jdbi = jdbi;
+    this.jdbi.registerArrayType(new LinkArrayType());
+    this.jdbi.registerArgument(new LinkArgumentFactory());
+    this.jdbi.registerRowMapper(new CollectionMapper());
+  }
+
   @Override
   public Response addCollection(Collection collection) {
-    throw new UnsupportedOperationException();
+    UUID collectionId = UUID.randomUUID();
+    jdbi.useHandle(
+        handle ->
+            handle
+                .createUpdate(
+                    "insert into collections (id, title, description, links, extent, item_type, crs) values (:id, :title, :description, CAST(:links AS jsonb[]), CAST(:extent AS jsonb), :item_type, :crs)") //
+                .bind("id", collectionId)
+                .bind("title", collection.getTitle())
+                .bind("description", collection.getDescription())
+                .bindArray("links", Link.class, collection.getLinks())
+                .bindByType("extent", collection.getExtent(), EXTENT)
+                .bind("item_type", collection.getItemType())
+                .bindArray("crs", String.class, collection.getCrs())
+                .execute());
+    return Response.created(URI.create("collections/" + collectionId)).build();
   }
 
   @Override
   public Response deleteCollection(UUID collectionId) {
-    throw new UnsupportedOperationException();
+    jdbi.useHandle(
+        handle -> handle.execute("delete from collections where id = (?)", collectionId));
+
+    return Response.noContent().build();
   }
 
   @Override
   public Response getCollection(UUID collectionId) {
-    throw new UnsupportedOperationException();
+    Collection collection =
+        jdbi.withHandle(
+            handle ->
+                handle
+                    .createQuery(
+                        "select id, title, description, links, extent, item_type, crs from collections where id = :id")
+                    .bind("id", collectionId)
+                    .map(new CollectionMapper())
+                    .one());
+
+    return Response.ok(collection).build();
   }
 
   @Override
   public Response getCollections() {
-    throw new UnsupportedOperationException();
+    List<Collection> collectionList =
+        jdbi.withHandle(
+            handle ->
+                handle
+                    .createQuery(
+                        "select id, title, description, links, extent, item_type, crs from collections")
+                    .map(new CollectionMapper())
+                    .list());
+
+    Collections collections = new Collections();
+    collections.setCollections(collectionList);
+
+    return Response.ok(collections).build();
   }
 
   @Override
   public Response updateCollection(UUID collectionId, Collection collection) {
-    throw new UnsupportedOperationException();
+    jdbi.useHandle(
+        handle ->
+            handle
+                .createUpdate(
```

Comment:
Have you considered binding the collection (`.bindPojo(collection)`)

Reply:
No, maybe we can have a look at this a some point. I am not sure how to facilitate this with the json types.